jetty-ee11/jetty-ee11-servlet/src/main/java/org/eclipse/jetty/ee11/servlet/ServletContextHolder.java

```diff
@@ -0,0 +1,163 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.ee11.servlet;
+
+import jakarta.servlet.ServletContext;
+import jakarta.servlet.UnavailableException;
+import org.eclipse.jetty.ee.BaseHolder;
+import org.eclipse.jetty.ee.Source;
+import org.eclipse.jetty.server.handler.ContextHandler;
+import org.eclipse.jetty.util.annotation.ManagedAttribute;
+import org.eclipse.jetty.util.annotation.ManagedObject;
+
+/**
+ * Specialization of BaseHolder for servlet-related classes
+ *
+ * @param <T> the type of holder
+ */
+@ManagedObject("Holder - a container for servlets and the like")
+public abstract class ServletContextHolder<T> extends BaseHolder<T>
```

Comment:
I think this name is misleading: it is not holding a `ServletContext`. Not sure of what a better name might be? `ServletContextScopedHolder`?